maths/leaky_relu.py

```diff
@@ -0,0 +1,58 @@
+"""
+This algorithm implements the leaky rectified linear algorithm (LReLU).
+
+LReLU is at times used as a substitute to ReLU because it fixes the dying ReLU problem.
+This is done by adding a slight slope to the negative portion of the function.
+The default value for the slope is 0.01.
+The new slope is determined before the network is trained.
+
+Script inspired from its corresponding Wikipedia article
+https://en.wikipedia.org/wiki/Rectifier_(neural_networks)
+"""
+from __future__ import annotations
+
+
+def leaky_relu(
+    vector: float | list[float], negative_slope: float = 0.01
+) -> float | list[float]:
```

Comment:
```suggestion
def leaky_relu(vector: np.ndarray, negative_slope: float = 0.01) -> np.ndarray:
```
I think just type hinting it as `np.ndarray` is fine for this function. Using `numpy` arrays is pretty standard when it comes to NN-related Python programming, and `numpy` functions that take in arrays generally also support scalars as well ("`array_like`")